<?xml version="1.0"?>
<!--Launches and paremeterizes the point cloud accumulator-->
<launch>
  <!--Launch the nodelet manager-->
  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen"/>
  <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
  <node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid pcl_manager" output="screen">
    <remap from="~input" to="/stereo/points2"/>
    <remap from="~output" to="/stereo/points2_voxelgrid"/>
    <rosparam>
            filter_field_name: z
            filter_limit_min: 0.01
            filter_limit_max: 10
            filter_limit_negative: False
            leaf_size: 0.05
          </rosparam>
  </node>
  <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
  <node pkg="rustbot_accumulate_point_clouds" type="accumulatepointcloud" name="accumulatepointcloud" output="screen">
    <remap from="input" to="/stereo/points2_voxelgrid"/>
    <remap from="map" to="map"/>
  </node>
</launch>
